File: extracted/ru/runlayer/tests/fixtures/agent_detection/samples/sample_09/agent.ts
/** Minimal LangChain.js ReAct agent (framework detection corpus). */

import { tool } from "@langchain/core/tools";
import { createReactAgent } from "@langchain/langgraph/prebuilt";
import { ChatOpenAI } from "@langchain/openai";
import { z } from "zod";

const getWeather = tool(
  async ({ city }: { city: string }) => `It's 72F and sunny in ${city}.`,
  {
    name: "get_weather",
    description: "Get the current weather for a city.",
    schema: z.object({ city: z.string() }),
  },
);

const add = tool(
  async ({ a, b }: { a: number; b: number }) => `${a + b}`,
  {
    name: "add",
    description: "Add two integers.",
    schema: z.object({ a: z.number(), b: z.number() }),
  },
);

async function main(): Promise<void> {
  const llm = new ChatOpenAI({ model: "gpt-4o-mini" });
  const agent = createReactAgent({ llm, tools: [getWeather, add] });
  const result = await agent.invoke({
    messages: [
      {
        role: "user",
        content: "What's the weather in Paris, and what is 21 + 21?",
      },
    ],
  });
  console.log(result.messages[result.messages.length - 1].content);
}

void main();

File: extracted/ru/runlayer/tests/fixtures/agent_detection/samples/sample_02/agent.py
"""Minimal LangGraph prebuilt ReAct agent (framework detection corpus)."""

import os

from langchain_core.tools import tool
from langchain_openai import ChatOpenAI
from langgraph.prebuilt import create_react_agent


@tool
def get_weather(city: str) -> str:
    """Get the current weather for a city."""
    return f"It's 72F and sunny in {city}."


@tool
def add(a: int, b: int) -> int:
    """Add two integers."""
    return a + b


def main() -> None:
    model = ChatOpenAI(model="gpt-4o-mini", api_key=os.environ.get("OPENAI_API_KEY", ""))
    agent = create_react_agent(
        model,
        tools=[get_weather, add],
        prompt="You are a helpful assistant. Use tools when needed.",
    )
    result = agent.invoke(
        {"messages": [("user", "What's the weather in Paris, and what is 21 + 21?")]}
    )
    print(result["messages"][-1].content)


if __name__ == "__main__":
    main()
